doc/BUILD.md

```diff
@@ -1,6 +1,71 @@
 Build Instructions
 ==================
 
+## 1. Platform
+* Ubuntu 18.04
+
+## 2. Build Steps
+### 2.1 ROS2
+* Install ROS2 with dependencies from the ROS2 Installation Page: https://index.ros.org/doc/ros2/Installation/
+* Install the latest distribution of ROS2 to support Navigation2 master.
+* ROS2 can be installed either from binary packages or build from source. Choose to build ROS2 from source to support Navigation2 master.
+
+### 2.2 Navigation2 Dependencies
+If ROS2 installed from source (Ignore if installed from binary packages)
+```console
+$ source ~/ros2_ws/install/local_setup.bash
+```
+Fetch, build and install navigation2 dependencies:
+```console
+$ mkdir -p ~/ros2_nav_dependencies_ws/src
+$ cd ~/ros2_nav_dependencies_ws
+$ wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/ros2_dependencies.repos
```

Comment:
2 lines down you have `<distro-name>` but here you don't list the distro - it should be done so that people when we're on the 'G' distro looking for the 'E' install scripts aren't getting the wrong dependencies or cloning the wrong repo. I'd suggest updating all of them to use the `ROS2_DISTRO` env var to populate so that its copy-paste compatible with any distro into the future

Reply:
I think you meant `ROS_DISTRO`. Can we set `ROS_DISTRO` to `master` for the above wget cmd?

Reply:
Building off master isn't really a typical use case unless you're developing, and I would suspect that someone would be able to understand if there's no `G` release, to look at master. I would like to make these directions futureproof and so we're not having tickets appear because the state of the `.repos` file in master isnt matching their `N` release and not building.

Reply:
Building off master is the target use case here. Your comments got me thinking to include steps to build for release branches (e.g- dashing-devel). I will add a commit after validating the right  steps for building branches other than master. Let me know if I got you wrong.

Reply:
If the target is building master, then the markdown file should be renamed to building master. I want to make sure its explicit about what version everything is being grabbed in if there's going to be semantics including `master` `dashing` which are both called out in the document.

`git clone` without `-b master`, you don't necessarily know what you're getting by default. The raw github repos file is similar, that will be updated as needed for the edge distribution and may not be appropriate for everyone's builds.

Reply:
Made changes to be more specific about branch, kindly review.